test/smoke/testApps/AgentExceptionAndDependency/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/SampleTestDependencyAndExceptionWithAgent.java

```diff
@@ -0,0 +1,24 @@
+package com.microsoft.applicationinsights.smoketest;
+
+import com.microsoft.applicationinsights.internal.schemav2.RemoteDependencyData;
+
+import org.junit.*;
+
+import static org.junit.Assert.*;
+
+@UseAgent(value="test")
+@WithDependencyContainers(@DependencyContainer(value="redis", portMapping="6379"))
+public class SampleTestDependencyAndExceptionWithAgent extends AiSmokeTest {
+
+	@Test
+	@TargetUri("/trackData")
+	public void testDependencyAndExceptionWithAgent() throws Exception {
+		assertEquals(2, mockedIngestion.getCountForType("RemoteDependencyData"));
```

Comment:
Why do we get 2 RDDs? I am not sure. 

Does 'new Jedis()' call is also captured as RDD?

Reply:
From the results, you are right.